### ISSUE TITLE
Markdown Image does not show up

### ISSUE THREAD

**Issue by daniel1v** [stefan-niedermann/nextcloud-deck#963]
**Describe the bug**
Markdown Image from card attachment is not shown in the app 

Define Image in markdown:
`![IMG](/apps/deck/cards/15/attachment/3)`
 

**Expected behavior**
image should be displayed like in the Web ui

**Screenshots**
Not available 

**Versions**
 - Nextcloud:  20.0.8
 - Nextcloud Deck:  1.2.7
 - Nextcloud Android: 3.15.1
 - Nextcloud Android Deck: 1.18.2

**Smartphone (please complete the following information):**
- Device:  pixel 4a 
- Android-Version:  11
  -  F-Droid
  

**Stacktrace**
Not available

**Comment by stefan-niedermann** [comment]
This has not yet been implemented yet.

Issues: Attachments which have been uploaded prior to Deck Server app 1.3.x will allow this kind of embedding, while it is still problematic because the whole image will be downloaded.
Attachments which have been uploaded with Deck server 1.3.x or later will be stored as files on the server. Downloading thumbnails will be possible but thumbnails need a different URL to the attachment. Detecting whether an attachment has been uploaded before Deck server 1.3.x or after, is *not possible* by the URL. Implementing this feature backwardcompatible is not easy therefore.

Possible solution ideas: Parse the remoteId of the attachment (we have this in a util), find the corresponding attachment in our local list of attachments and get a thumbnail URL for it (this is also already available). If a thumbnail URL has been found, replace the link.

Further problems are relative vs. absolute links... So, no easy topic, but doable.

**Comment by stefan-niedermann** [comment]
Closing this issue as it **will be solved implicitly** when
- [x] the [`sso-glide`](https://github.com/stefan-niedermann/nextcloud-commons/#sso-glide) module of [`nextcloud-commons`](https://github.com/stefan-niedermann/nextcloud-commons/) supports both, absolute paths and absolute URLs to a Nextcloud instance and can delegate handling of public images to the default Glide image loader (fixed in `nextcloud-commons` version [`1.3.0`](https://github.com/stefan-niedermann/nextcloud-commons/releases/tag/1.3.0))
- [ ] the Markdown editor (used from the [Notes app](https://github.com/stefan-niedermann/nextcloud-notes/)) uses `sso-glide` to fetch images (currently the editor only supports embedding public images, because requesting an image of Nextcloud instance requires authentication)
  - Tracked in https://github.com/stefan-niedermann/nextcloud-notes/issues/1018
  - [ ] this step requires some support (or maybe changes) from the upstream [Markwon](https://noties.io/Markwon/) editor
    - Tracked in https://github.com/noties/Markwon/issues/329#issuecomment-850846313
- [ ] the Deck Android app has updated to the new Markdown editor
  - Doesn't need to be tracked, it will happen automatically by @dependabot-bot 
- [ ] the Deck server app has migrated all Deck attachments (uploaded prior to `1.3.0`) to File attachments (introduced with `1.3.0`)
  - This step is *optional* and if all other tasks are finished, we will already support displaying images which have been uploaded with Deck server `1.3.0` or later.
  - Basically the path to the attachment will change from `/apps/deck/cards/15/attachment/3` to `/index.php/core/preview?fileId=196163&x=1920&y=1080&a=true`
  - Tracked in https://github.com/nextcloud/deck/issues/3101

**Comment by stefan-niedermann** [comment]
Good news, we will ship support for all kind of Nextcloud-hosted images (Shared links, Preview links, File links, ...) as well as old Deck attachments (regardless of https://github.com/nextcloud/deck/issues/3101) in the next release (`1.18.4`  or `1.19.0` - we'll see). You will not need to change your URLs or anything, it will just work.
